python/examples/basic/Getting_Started_with_UDFs.ipynb

```diff
@@ -0,0 +1,211 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Table of Content"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we'll explore the basics of logging data with whylogs and a user defined function or UDFs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Installing whylogs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install whylogs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading a Pandas DataFrame"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before showing how we can log data, we first need the data itself. Let's create a simple Pandas DataFrame:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "data = {\n",
+    "    \"animal\": [\"cat\", \"hawk\", \"clam\", \"cat\", \"mongoose\", \"octopus\"],\n",
+    "    \"legs\": [4, 2, 1, 4, 4, 8],\n",
+    "    \"weight\": [4.3, 1.8, 1.3, 4.1, 5.4, 3.2],\n",
+    "}\n",
+    "\n",
+    "df = pd.DataFrame(data)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining a simple metric UDF\n",
+    "Here we use a metric UDF targeting a named column `animal` as an example to show how we can add features to a dataframe for custom monitoring. In this example we model some custom logic for if the animal has a cool name. This is a toy example that just checks if the name is longer than 4 characters, and does a binary classification, but you could return a score based on values in a column too."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import whylogs as why\n",
+    "from whylogs.experimental.core.udf_schema import udf_schema\n",
+    "from whylogs.experimental.core.metrics.udf_metric import register_metric_udf\n",
+    "\n",
+    "\n",
+    "@register_metric_udf(col_name=\"animal\")\n",
+    "def has_cool_animal_name(text):\n",
+    "  if len(text) > 4: # long names are cool\n",
+    "    return 1\n",
+    "  else:\n",
+    "    return 0\n",
+    "  \n",
+    "custom_schema = udf_schema()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Profiling with whylogs + UDFs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To obtain a profile of your data, you can simply use whylogs' `log` call with your UDF schema defined earlier. This will attach a feature named `animal.has_cool_animal_name` which you can then see in WhyLabs."
```

Comment:
This sounds like you're aiming for a dataset UDF rather than a `UdfMetric`?

Reply:
Its a description of what they will see in WhyLabs, a feature named `animal.has_cool_animal_name`